### DISCUSSION TITLE
How to view/debug/browse the AST

### DISCUSSION THREAD

**Discussion by BjAlvestad** [eclipse-langium/langium#1173] [Q&A] [answered]
I am a complete newbie to this, so apologies if it is a silly question.

Is there some easy way to view the resulting AST from parsing one of my test files for debug purpose?
The reason I wish to view the AST is help me resolve issues like
"Could not resolve reference to block named 'myBlockId'. myLanguage (linking-error)"

I believe it would be helpful for me to see if `myBlockId` even exist in the AST, and to view the actual relative position of them to see if the issue is due to default scoping rules, or if it did not get properly parsed at all.

Feel free to also suggest alternative ways of debugging.
My main issue is that I feel like I am flying blind, since I lack the know how of how to meaningfully debug by grammar / code / resulting AST.

**Comment by Yokozuna59** [comment]
Hi there, @BjAlvestad, Just to clarify, I'm not part of the Langium team, so please don't answer as the only choice; you could create your own [playground](https://langium.org/playground/).
I'm not sure how hard it is to implement such a thing, but it should feasible. You might want to take a look at [showcases](https://langium.org/showcase/) for more ideas.

**Comment by spoenemann** [comment, marked as answer]
The [Playground](https://langium.org/playground/) is the quickest way to show parsed ASTs.

To inspect ASTs right from your language server / app, use the [JsonSerializer](https://github.com/eclipse-langium/langium/blob/6145872ce057aad9c9bfb4ffbdbe9c817bbaf82c/packages/langium/src/serializer/json-serializer.ts#L67-L78) service. For example, you can write the result to a file and inspect the content with an editor or other tool.
